feat: add IOTA-Names CLI and infrastructure

thibault-martinez opened the pull request:
### Release Notes

- [ ] Protocol:
- [x] Nodes (Validators and Full nodes): Added an optional `IotaNamesConfig` option to the `NodeConfig`
- [x] Indexer: Added `iotax_iotaNamesLookup`, `iotax_iotaNamesReverseLookup`, `iotax_iotaNamesFindAllRegistrationNFTs` RPC endpoints (Devnet support only). Added `iota-names` configuration options to the `IndexerConfig`.
- [x] JSON-RPC: Added `iotax_iotaNamesLookup`, `iotax_iotaNamesReverseLookup`, `iotax_iotaNamesFindAllRegistrationNFTs` RPC endpoints (Devnet support only).
- [x] GraphQL: Added `resolveIotaNamesAddress` query and `iotaNamesDefaultName` field for address (Devnet support only). Added `iota-names` configuration options to the `ServiceConfig`.
- [x] CLI: Added `iota name` command to operate IOTA-Names smart contracts deployed on the devnet only
- [ ] Rust SDK:
- [ ] REST API:

DaughterOfMars committed:
feat: Add `iota name` command (#71)

* feat: Add `iota name` command

* :triumph:

* license

* remove humantime

* simplify types

* revert opts clone

* what year is it?

* review

* add expect tag for now

* better regex

* refactor get objects

* add prompt

Thoralf-M committed:
feat(iota): add NameCommand::Transfer (#77)

* feat(iota): add NameCommand::Transfer

* Apply suggestions from code review

Co-authored-by: DaughterOfMars <chloedaughterofmars@gmail.com>

---------

Co-authored-by: DaughterOfMars <chloedaughterofmars@gmail.com>

Thoralf-M committed:
feat(iota): add NameCommand::Burn (#78)

* feat(iota): add NameCommand::Burn

* feat(iota): add expiration check to burn

* order!

* Update crates/iota/src/name_commands.rs

* Update crates/iota/src/name_commands.rs

Co-authored-by: DaughterOfMars <chloedaughterofmars@gmail.com>

---------

Co-authored-by: Thibault Martinez <thibault@iota.org>
Co-authored-by: DaughterOfMars <chloedaughterofmars@gmail.com>

thibault-martinez committed:
feat: add iota name list command (#80)

* feat: add `iota name list` command

* factorize

* nits

* address parameter

* SystemTime utilities

* add expiration to burn error message

* revert

* nits

* Update crates/iota/src/name_commands.rs

Co-authored-by: Thoralf-M <46689931+Thoralf-M@users.noreply.github.com>

---------

Co-authored-by: Thoralf-M <46689931+Thoralf-M@users.noreply.github.com>

DaughterOfMars committed:
fix: Serialize address option as vec (#86)

Thoralf-M committed:
feat(iota): add NameCommand::ReverseLookup (#82)

* feat: add NameCommand::ReverseLookup

* feat(iota): make ReverseLookup address optional and use active address as default

* only print the domain name

* Update comments

Co-authored-by: /alex/ <alexander.schmidt@iota.org>

* fmt

* simplify getting the name

* Apply suggestions from code review

Co-authored-by: /alex/ <alexander.schmidt@iota.org>

* refactor: use bcs instead of json value

* feat(iota): expect(dead_code)

* Apply suggestions from code review

---------

Co-authored-by: /alex/ <alexander.schmidt@iota.org>

thibault-martinez committed:
feat(iota): add `NameCommand::{SetReverseLookup, UnsetReverseLookup}` (#90)

* feat(iota): add `NameCommand::SetReverseLookup`

* add UnsetReverseLookup and check ownership of the NFT

* add comment

Alex6323 committed:
feat(iota): add `NameCommand::Lookup` (#84)

* impl lookup command

* review 1

* docs

Co-authored-by: Thibault Martinez <thibault@iota.org>

* keep things private

* use bcs

* better print

* fix warnings

* align + serde renames

* nit

* rm renames

---------

Co-authored-by: Thibault Martinez <thibault@iota.org>

DaughterOfMars committed:
feat: Add set/unset user data commands (#91)

* feat: Add set/unset user data commands

* comments

* Add get-user-data

* clippy

Alex6323 committed:
feat(iota): add `NameCommand::Register` (#79)

* feat: Add `iota name` command

* :triumph:

* license

* remove humantime

* simplify types

* revert opts clone

* what year is it?

* review

* add expect tag for now

* better regex

* refactor get objects

* add prompt

* feat: Add `iota name` command

* remove duplicate length validation

* review: rename to register

* fix merge

* fix merge 2

* fetch price from network

* review suggestion

Co-authored-by: DaughterOfMars <chloedaughterofmars@gmail.com>

* use iota-graphql-rpc-client

* fix quotes

* segment -> label

* use get_dynamic_field_object instead of graphql

* fix query

* improvements

* improvements 2

* fix regex

* provide coin for payment

* improvements 3

* domain->name for register cmd

* domain->name for all

* self review

* optional coin

* rm evidence

* append args

* simplify

* fix regex

* use bcs

* rm evidence

* some renaming

* fix clippy

* undo rename

* undo rename 2

* undo rename 3

* review suggestions 1

Co-authored-by: DaughterOfMars <chloedaughterofmars@gmail.com>

* undo rename 4

* consistence 1

* not this

* consistence 2

---------

Co-authored-by: Chloe Martin <chloedaughterofmars@gmail.com>

Thoralf-M committed:
feat(iota): remove obsolete clap rename (#101)

DaughterOfMars committed:
feat: Add `NameCommandResult` (#92)

* feat: Add `NameCommandResult`

* Print above

* rename

* review

* Use trait more

* nits

* combine UserData variants

* only return keyed value

---------

Co-authored-by: Thibault Martinez <thibault@iota.org>

thibault-martinez committed:
fix(iota): `prompt_if_no_config` iota-names compilation (#105)

* fix(iota): `prompt_if_no_config` iota-names compilation

* nit

thibault-martinez committed:
feat: add `iota-names` crate (#106)

* feat: add `iota-names` crate

* move constants

* reorder things

* move registry entries

* remove mainnet/testnet config

* Modifications Copyright

* cargo sort

* remove unused error variant

* use `NameServiceConfig` to compute dynamic field ID locally

* add config module

* add domain and error modules

* add registry module

* move TODO

* Update crates/iota-names/Cargo.toml

Co-authored-by: Thoralf-M <46689931+Thoralf-M@users.noreply.github.com>

* reenable parent_extraction

* rename error to IotaNamesError

* fix(iota-names): MAX_DOMAIN_LENGTH

* fix(iota-names): add TODO

* set CODEOWNERS for iota-names/

* clippy

* improve comment

Co-authored-by: DaughterOfMars <chloedaughterofmars@gmail.com>

* add iota-names to crates-filters.yml

---------

Co-authored-by: Thoralf-M <46689931+Thoralf-M@users.noreply.github.com>
Co-authored-by: Thoralf Müller <thoralf.mue@gmail.com>
Co-authored-by: DaughterOfMars <chloedaughterofmars@gmail.com>

thibault-martinez committed:
feat: add `iota-names` crate (#106)

* feat: add `iota-names` crate

* move constants

* reorder things

* move registry entries

* remove mainnet/testnet config

* Modifications Copyright

* cargo sort

* remove unused error variant

* use `NameServiceConfig` to compute dynamic field ID locally

* add config module

* add domain and error modules

* add registry module

* move TODO

* Update crates/iota-names/Cargo.toml

Co-authored-by: Thoralf-M <46689931+Thoralf-M@users.noreply.github.com>

* reenable parent_extraction

* rename error to IotaNamesError

* fix(iota-names): MAX_DOMAIN_LENGTH

* fix(iota-names): add TODO

* set CODEOWNERS for iota-names/

* clippy

* improve comment

Co-authored-by: DaughterOfMars <chloedaughterofmars@gmail.com>

* add iota-names to crates-filters.yml

---------

Co-authored-by: Thoralf-M <46689931+Thoralf-M@users.noreply.github.com>
Co-authored-by: Thoralf Müller <thoralf.mue@gmail.com>
Co-authored-by: DaughterOfMars <chloedaughterofmars@gmail.com>

Thoralf-M committed:
feat(iota/iota-names): get NameServiceConfig based on active network (#110)

* feat(iota/iota-names): get NameServiceConfig based on active network

* accept Chain instead of identifier

* apply review suggestions

* more renames

* address review comments

DaughterOfMars committed:
feat: Filter `PrintableResult` using `should_print` (#107)

Thoralf-M committed:
feat(iota, iota-names)!: update IDs, PTBs and config to new version (#113)

* feat(iota, iota-names)!: update IDs, PTBs and config to new version

* rename

* use valid digest

thibault-martinez committed:
feat(iota-names): add `IotaNamesRegistration::type_` (#112)

* feat(iota-names): add `IotaNamesRegistration::type_`

* move constants to function

thibault-martinez committed:
chore(iota-names): use IDs from iota-types (#111)

thibault-martinez committed:
chore(cli): better error on `get_reverse_registry_entry` (#114)

* better error on get_reverse_registry_entry

* clippy :sad-cat:

thibault-martinez committed:
chore(cli): local compute pricing config object ID (#115)

Thoralf-M committed:
feat(iota): add name renewal command (#118)

* feat(iota): add name renewal command

* Address review comments

thibault-martinez committed:
feat(cli): add name auction subcommand (#117)

* feat(cli): add name auction subcommand

* add get_auction_house

* impl start

* impl bid

* impl claim

* :sad-cat:

* impl metadata

* remove get_auction_house

* better metadata output

* docs

* better metadata format

* Update crates/iota/src/name_commands.rs

Co-authored-by: Thoralf-M <46689931+Thoralf-M@users.noreply.github.com>

* update IDs

* remove auction from iota-names crate

* remove unwraps

* Rename NameCommandResult::Metadata

* Update crates/iota/src/name_commands.rs

Co-authored-by: Thoralf-M <46689931+Thoralf-M@users.noreply.github.com>

* doc comment

* TODO

---------

Co-authored-by: Thoralf-M <46689931+Thoralf-M@users.noreply.github.com>

Thoralf-M committed:
feat(iota): accept amount for auction start/bid commands (#126)

DaughterOfMars committed:
feat: Add subdomain commands (#76)

* feat: Add subdomain commands

* doc

* fix subdomains_package usage, list also subdomain nodes

* optional address in register leaf

* better timestamp support

* rename len

* review

* comment

* review and improve timestamp parsing

* update comments

* make it optional

* fix merge

* use subdomain proxy

* remove humantime

* fix doc test

* fix metadata update

* fix renew

* review

* being consistent

* revert

* refactor to dedup

* remove IDE artefact

---------

Co-authored-by: Thoralf Müller <thoralf.mue@gmail.com>
Co-authored-by: Thibault Martinez <thibault@iota.org>

Thoralf-M committed:
feat(iota): fetch auction IDs dynamically (#120)

* feat(iota): fetch auction IDs dynamically

* refactor, rename function

* refactor

* Remove unnecessary generic types

Thoralf-M committed:
fix(iota-names): update package IDs (#131)

Commit:
feat(iota-json-rpc-api)!: Add IOTA-Names Indexer and Node RPCs (#116)

* feat: Add Indexer and Node RPCs

* refactor: Return IotaNameRecord

* refactor: Fix dprint

* refactor: Fix blocking tasks

* refactor: Regenerate OpenAPI spec

* refactor: Fix error annotation

Co-authored-by: Thoralf-M <46689931+Thoralf-M@users.noreply.github.com>

* refactor: Refactor function signatures, improve error handling

* refactor: Rename to `iota_names_config` and regenerate test snapshots

* feat: Add API to find all registration NFTs

* refactor: Refactor injection of config

* refactor: Remove unused dep, format

* refactor: Improve config handling

* refactor: Fix wrong clap rename

* refactor: Fix nit

* refactor: Update OpenRPC spec

* refactor: Simplify and improve `iota_names_resolve_record`

* refactor: Clarify iota-json-rpc `iota_names_resolve_record`, fix iota-indexer error variant

* refactor: Use get_owned_objects instead RPC instead get_owned_objects_internal

* Add serde default for config

Co-authored-by: Konstantinos Demartinos <konstantinos.demartinos@iota.org>

* refactor: Remove unused error variants; nits

* refactor: Clippy

* refactor: Simplify subdomain check

* refactor: Pass `&str` instead of `String`

* refactor: Remove unncessary deps; fix license header

* refactor: Fix license header

* refactor: Add missing serde rename

* refactor: Fix namings

* refactor: Fix base branch changes

* fix: Rename to iota_names_lookup

* Update crates/iota-json-rpc/src/indexer_api.rs

* Update crates/iota-indexer/src/apis/indexer_api.rs

---------

Co-authored-by: Thoralf-M <46689931+Thoralf-M@users.noreply.github.com>
Co-authored-by: Konstantinos Demartinos <konstantinos.demartinos@iota.org>
Co-authored-by: Thibault Martinez <thibault@iota.org>

Commit:
feat(iota-graphql-rpc): Add IOTA-Names GraphQL endpoints (#119)

* feat: WIP

* refactor: Make the GraphQL upstream changes compile

* refactor: Refactor naming

* refactor: Wire resolve_iota_names_address query

* refactor: Nits

* refactor: Adjust header

* refactor: Nits and fixing rename

* refactor: Revert devnet config changes

* refactor: Rename `default_iota_names_name` function

* fix: Fix merge conflict

* fix: Update graphql example

* fix: Fix merge conflict

* fix: Typo

Co-authored-by: Thibault Martinez <thibault@iota.org>

---------

Co-authored-by: Thibault Martinez <thibault@iota.org>

thibault-martinez committed:
fix: add display field to IOTA-Names PTBs (#138)

* fix: add display field to IOTA-Names PTBs

* Default::default

muXxer commented on crates/iota-names/src/registry.rs:
```diff
@@ -0,0 +1,137 @@
+    pub id: ObjectID,
+    pub size: u64,
+
+    // TODO: remove these fields and <K, V> if they aren't needed for the indexer
```
should we add an issue here to track it?

thibault-martinez replied:
Done in https://github.com/iotaledger/iota/pull/6530

muXxer commented on crates/iota-names/src/error.rs:
```diff
@@ -0,0 +1,25 @@
+    #[error("Name Service: Hyphens are not allowed as the first or last character")]
+    InvalidHyphens,
+    #[error("Name Service: Only lowercase letters, numbers, and hyphens are allowed")]
+    InvalidUnderscore,
```
The error message sounds like no special characters are allowed, but the name of the error itself sounds more like only underscores are not allowed.

thibault-martinez replied:
Changed in https://github.com/iotaledger/iota/pull/6530

muXxer commented on crates/iota-names/src/domain.rs:
```diff
@@ -0,0 +1,384 @@
+//         }
+//         Ok(Self { labels })
+//     }
+// }
```
what is that here? the corresponding move code?
Maybe we could at least add a comment why this is here, or remove it completely?

thibault-martinez replied:
Removed in https://github.com/iotaledger/iota/pull/6558

muXxer commented on crates/iota-names/src/domain.rs:
```diff
@@ -0,0 +1,384 @@
+//         "invalid characters in domain: {label}"
+//     );
+//     Ok(label.to_owned())
+// }
```
dead code? remove it or at least add a comment/issue to enable it.

thibault-martinez replied:
Was removed https://github.com/iotaledger/iota/pull/6530

muXxer commented on crates/iota-names/src/domain.rs:
```diff
@@ -0,0 +1,384 @@
+        let _tld = labels.pop();
+        let sld = labels.pop().unwrap();
+
+        format!("{}{}{}", labels.join(sep), IOTA_NEW_FORMAT_SEPARATOR, sld)
```
rust noob here, but I don't get that line. what is the output supposed to look like?
isn't the TLD gone in that case?

thibault-martinez replied:
Yes it is gone, there is no TLD in the AT format, it would still be there in the DOT format that is done just above

muXxer replied:
maybe add a comment above that line how the output is supposed to look like?

thibault-martinez replied:
Done in https://github.com/iotaledger/iota/pull/6558

muXxer commented on crates/iota-names/src/domain.rs:
```diff
@@ -0,0 +1,384 @@
+}
+
+/// Parses a separator from the domain string input.
+/// E.g.  `example.iota` -> `.` | example*iota -> `@` | `example*iota` -> `*`
```
example*iota -> `@`

that comment seems wrong.

thibault-martinez replied:
Function removed in https://github.com/iotaledger/iota/pull/6558

muXxer commented on crates/iota-names/src/domain.rs:
```diff
@@ -0,0 +1,384 @@
+    };
+
+    let Some(after) = splits.next() else {
+        return Ok(before.to_string());
```
I don't get that one. Shouldn't we still add DEFAULT_TLD and the separator in that case?

thibault-martinez replied:
If there is no `after`, then it was not in the AT format and it already contains the TLD

muXxer commented on crates/iota-names/src/domain.rs:
```diff
@@ -0,0 +1,384 @@
+    Ok(parts.join(&separator.to_string()))
+}
+
+pub fn validate_label(label: &str) -> Result<&str, IotaNamesError> {
```
I would mention in a comment here what a valid label is.

length 1-63, a-z, 0-9, "-" not allowed at the start or end.

thibault-martinez replied:
Done in https://github.com/iotaledger/iota/pull/6530

muXxer commented on crates/iota-names/src/domain.rs:
```diff
@@ -0,0 +1,384 @@
+        if !is_valid_character {
+            match character {
+                b'-' => return Err(IotaNamesError::InvalidHyphens),
+                _ => return Err(IotaNamesError::InvalidUnderscore),
```
Ah I think you mixed the rust "_" with the ascii Underline.... at least the name of the error makes no sense to me, because we are not only matching against underlines here, but all characters that are not represented by the above `is_valid_character` check.

thibault-martinez replied:
No this is actually correct, but I do agree the error name is not ideal, will change

thibault-martinez replied:
Changed in https://github.com/iotaledger/iota/pull/6530

muXxer replied:
Yeah I didn't want to say that the code is wrong. The code is correct with the `_` case, but the error name was wrong because it wasn't about matching against the character "_" which the error name referred to.

thibault-martinez replied:
Yeah I agree, something coming from upstream that we overlooked

muXxer commented on crates/iota-names/src/config.rs:
```diff
@@ -0,0 +1,120 @@
+#[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
+#[serde(rename_all = "kebab-case")]
+pub struct IotaNamesConfig {
+    pub package_address: IotaAddress,
+    pub object_id: ObjectID,
+    pub payment_package_address: IotaAddress,
+    pub registry_id: ObjectID,
+    pub reverse_registry_id: ObjectID,
+    pub subdomain_proxy_package_id: ObjectID,
```
Can't we add some comments to those fields? For me they are not really obvious if I look at the IOTA Name stuff for the first time.

thibault-martinez replied:
Added in https://github.com/iotaledger/iota/pull/6530

muXxer commented on crates/iota-names/src/config.rs:
```diff
@@ -0,0 +1,120 @@
+            // Chain::Mainnet => IotaNamesConfig::mainnet(),
+            // Chain::Testnet => IotaNamesConfig::testnet(),
+            Chain::Unknown => IotaNamesConfig::devnet(),
+            _ => todo!("uncomment Mainnet and Testnet when IOTA-Names is published there"),
```
we should add the tracking issue here.

thibault-martinez replied:
Done in https://github.com/iotaledger/iota/pull/6530

muXxer commented on crates/iota-names/src/config.rs:
```diff
@@ -0,0 +1,120 @@
+    }
+
+    pub fn record_field_id(&self, domain: &Domain) -> ObjectID {
+        let domain_type_tag = Domain::type_(self.package_address);
```
why does it point to the package_address and not to domain here?
I don't understand that, maybe add a comment here?

thibault-martinez replied:
We're constructing the struct tag so we need the package address, see `type_`'s implementation

muXxer commented on crates/iota-names/src/config.rs:
```diff
@@ -0,0 +1,120 @@
+        .unwrap()
+    }
+
+    // TODO add mainnet() and testnet()
```
add the tracking issue.

thibault-martinez replied:
Done in https://github.com/iotaledger/iota/pull/6530

muXxer commented on crates/iota-names/src/lib.rs:
```diff
@@ -0,0 +1,133 @@
+    }
+
+    fn expiration_time(&self) -> SystemTime {
+        UNIX_EPOCH + Duration::from_millis(self.expiration_timestamp_ms())
```
Oh I thought the expiration_timestamp_ms is already in unix time? The UNIX_EPOCH here confuses me a bit. So it's not? Or is that only needed to make a SystemTime object out of it?

thibault-martinez replied:
Not sure to follow, it is a unix timestamp, which is the number os s/ms since the unix epoch, so if you want the actual date you need to add both

muXxer replied:
yeah never mind.
the `expiration_timestamp_ms()` are the milliseconds since UNIX_EPOCH.
for creating a valid `SystemTime` you need to parse the duration from those millis and add the `UNIX_EPOCH`.
I was just a bit confused while reading it, because I never handled unix timestamps in rust yet.

thibault-martinez committed:
cli: feature-gated name command (#6526)

thibault-martinez committed:
chore(iota-names): some cleanup and improvements (#6530)

Co-authored-by: Thoralf-M <46689931+Thoralf-M@users.noreply.github.com>

samuel-rufi commented:
Should wait for https://github.com/iotaledger/iota/pull/6625 to be merged

Commit:
refactor(iota-indexer): Use `IotaNamesConfig::default()` if not populated in `NodeConfig` (#6625)

# Description of change

In case the optional `IotaNamesConfig` isn't populated in the
`NodeConfig` the node needs to instantiate the correct `IotaNamesConfig`
itself. As done currently `IotaNamesConfig::from_chain` is used for that
which however isn't fully implemented yet and would run into panic, as
the `testnet/mainnet` branches are not implemented.

Therefore this PR replaces that logic for now to use
`IotaNamesConfig::default()` if no config is provided. This aligns with
the default configuration for the Indexer and GraphQL services.

In future, this change should be reverted so that the node will be able
to derive the `IotaNamesConfig` directly from the chain, similarly as
done in upstream
(https://github.com/MystenLabs/sui/blob/main/crates/sui-node/src/lib.rs#L2247-L2251).
This would bring out-of-the-box support for `Chain::{Unknown,Testnet,
Mainnet}` without the need of providing a configuration to the node.

## Type of change

- Feature (Non-breaking change)

## How the change has been tested

- [X] Basic tests (linting, compilation, formatting, unit/integration
tests)
- [X] Patch-specific tests (correctness, functionality coverage)

### Infrastructure QA (only required for crates that are maintained by
@iotaledger/infrastructure)

- [ ] Synchronization of the indexer from genesis for a network
including migration objects.
- [ ] Restart of indexer synchronization locally without resetting the
database.
- [ ] Restart of indexer synchronization on a production-like database.
- [X] Deployment of services using Docker.
- [X] Verification of API backward compatibility.

## Change checklist

- [X] I have followed the contribution guidelines for this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [X] I have checked that new and existing unit tests pass locally with
my changes

### Release Notes

<!--
Check each box that your changes affect. If none of the boxes relate to
your changes, release notes aren't required.

For each box you select, include information after the relevant heading
that describes the impact of your changes that a user might notice and
any actions they must take to implement updates.
-->

- [ ] Protocol:
- [ ] Nodes (Validators and Full nodes):
- [ ] Indexer:
- [ ] JSON-RPC:
- [ ] GraphQL:
- [ ] CLI:
- [ ] Rust SDK:
- [ ] REST API:

thibault-martinez committed:
chore(iota-names): domain cleanup (#6558)

Improves errors, removes outdated and commented code, removes unwanted
`*` syntax, increases reusability.

thibault-martinez committed:
fix(iota): clippy warning in names command (#6646)

thibault-martinez committed:
chore: remove `aws-smithy-http` udep (#6652)

Simply removed an unused dependency.

Commit:
refactor(iota-graphql-rpc): Regenerate GraphQL spec and update test snapshot file (#6655)

# Description of change

Updates the GraphQL spec and test snapshot file to accommodate
https://github.com/iotaledger/iota/pull/6523. Also fixes the Cargo.lock
file.

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

`cargo nextest run -p iota-graphql-rpc --features pg_integration
--no-fail-fast --test-threads 1`

- [X] Basic tests (linting, compilation, formatting, unit/integration
tests)
- [X] Patch-specific tests (correctness, functionality coverage)

### Infrastructure QA (only required for crates that are maintained by
@iotaledger/infrastructure)

- [ ] Synchronization of the indexer from genesis for a network
including migration objects.
- [ ] Restart of indexer synchronization locally without resetting the
database.
- [ ] Restart of indexer synchronization on a production-like database.
- [ ] Deployment of services using Docker.
- [X] Verification of API backward compatibility.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items
that are not.

- [X] I have followed the contribution guidelines for this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [X] I have checked that new and existing unit tests pass locally with
my changes

### Release Notes

<!--
Check each box that your changes affect. If none of the boxes relate to
your changes, release notes aren't required.

For each box you select, include information after the relevant heading
that describes the impact of your changes that a user might notice and
any actions they must take to implement updates.
-->

- [ ] Protocol:
- [ ] Nodes (Validators and Full nodes):
- [ ] Indexer:
- [ ] JSON-RPC:
- [ ] GraphQL:
- [ ] CLI:
- [ ] Rust SDK:
- [ ] REST API:

---------

Co-authored-by: Thibault Martinez <thibault@iota.org>

thibault-martinez committed:
chore: update Cargo.lock (#6669)
